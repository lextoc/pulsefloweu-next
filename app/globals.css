.divider {
  margin: 3rem 0;
}

.content {
  margin-bottom: 25vh;
}

.cards {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(16rem, 20rem));
  gap: 1.5rem;
}
