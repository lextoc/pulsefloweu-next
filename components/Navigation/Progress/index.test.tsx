import { expect } from "@jest/globals";
import { useIsFetching } from "@tanstack/react-query";
import { render } from "@testing-library/react";
import React from "react";

import NavigationProgress, { INavigationProgressProps } from "./index";

// Mock @tanstack/react-query module
jest.mock("@tanstack/react-query", () => ({
  useIsFetching: jest.fn(),
}));

describe("NavigationProgress component", () => {
  const mockUseIsFetching = useIsFetching as jest.Mock;

  it("applies isFetching styles when isFetching is true", () => {
    mockUseIsFetching.mockReturnValue(true);

    const { container } = render(<NavigationProgress />);

    const navigationProgressElement =
      container.querySelector(".root.isFetching");

    expect(navigationProgressElement).toBeInTheDocument();
  });

  it("does not apply isFetching styles when isFetching is false", () => {
    mockUseIsFetching.mockReturnValue(false);

    const { container } = render(<NavigationProgress />);

    const navigationProgressElement =
      container.querySelector(".root.isFetching");

    expect(navigationProgressElement).toBeNull();
  });
});
