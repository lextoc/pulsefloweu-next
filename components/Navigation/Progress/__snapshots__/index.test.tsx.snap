// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationProgress component matches snapshot 1`] = `
<div>
  <div
    class="root "
  />
</div>
`;
